statsmodels/tsa/statespace/regime_switching/kim_filter.py

```diff
@@ -0,0 +1,635 @@
+        return self.regime_filters[0].initial_variance
+
+    @initial_variance.setter
+    def initial_variance(self, value):
+        for regime_filter in self.regime_filters:
+            regime_filter.initial_variance = value
+
+    @property
+    def loglikelihood_burn(self):
+        return self._loglikelihood_burn
+
+    @loglikelihood_burn.setter
+    def loglikelihood_burn(self, value):
+        self._loglikelihood_burn = value
+        for regime_filter in self.regime_filters:
+            regime_filter.loglikelihood_burn = value
+
+    @property
+    def tolerance(self):
+        return self.regime_filters[0].tolerance
+
+    @tolerance.setter
+    def tolerance(self, value):
+        for regime_filter in self.regime_filters:
+            regime_filter.tolerance = value
+
+    @property
+    def _complex_endog(self):
+        return self.regime_filters[0]._complex_endog
+
+    def _initialize_filter(self, **kwargs):
+
+        kwargs['filter_timing'] = 1
+
+        for regime_filter in self._regime_kalman_filters:
+            regime_filter._initialize_filter(**kwargs)
+
+    def set_filter_method(self, **kwargs):
+
+        for regime_filter in self._regime_kalman_filters:
+            regime_filter.set_filter_method(**kwargs)
+
+    def set_inversion_method(self, **kwargs):
+
+        for regime_filter in self._regime_kalman_filters:
+            regime_filter.set_inversion_method(**kwargs)
+
+    def set_stability_method(self, **kwargs):
+
+        for regime_filter in self._regime_kalman_filters:
+            regime_filter.set_stability_method(**kwargs)
+
+    def set_conserve_memory(self, **kwargs):
+
+        for regime_filter in self._regime_kalman_filters:
+            regime_filter.set_conserve_memory(**kwargs)
+
+    def bind(self, endog):
+
+        for regime_filter in self._regime_kalman_filters:
+            regime_filter.bind(endog)
+
+        self._nobs = self._regime_kalman_filters[0].nobs
+
+    def initialize_known(self, initial_states, initial_state_covs):
+
+        k_regimes = self._k_regimes
+        regime_filters = self._regime_kalman_filters
+
+        initial_states = self._prepare_data_for_regimes(initial_states, 1)
+        initial_state_covs = self._prepare_data_for_regimes(initial_state_covs,
+                2)
+
+        for i in range(k_regimes):
+            regime_filters[i].initialize_known(initial_states[i],
+                    initial_state_covs[i])
+
+    def initialize_stationary(self):
+
+        for regime_filter in self._regime_kalman_filters:
+            regime_filter.initialize_stationary()
+
+    def initialize_known_regime_probs(self, initial_regime_probs):
+
+        self._initial_regime_probs = initial_regime_probs
+
+    def initialize_uniform_regime_probs(self):
+
+        k_regimes = self._k_regimes
+        self._initial_regime_probs = np.ones((k_regimes,), dtype=self._dtype) \
+                / k_regimes
+
+    def initialize_stationary_regime_probs(self):
```

Comment:
Once you've constructed `a_eq`, the steady state probabilities can be found as: `np.linalg.pinv(a_eq)[:, -1]`. I think that if the non-negativity constraints are binding, the problem is not well specified anyway (i.e. some regimes cannot be reached).